### PR TITLE
Release v0.3: update ibc-go to v4.0.0

### PR DESCRIPTION
**Pull request by bluele**
implement #73

This PR should be merged after release v0.2(#69) 

Signed-off-by: Jun Kimura <jun.kimura@datachain.jp>

### CODE REVIEW

**Comment by bluele** [comment]
[v4](https://github.com/cosmos/ibc-go/releases/tag/v4.0.0) is available now. We should upgrade to this.

**Comment by bluele** [comment]
ibc-go v4.0.0 is already released
https://github.com/cosmos/ibc-go/releases/tag/v4.0.0

File: chains/tendermint/light.go [hunk 1]
```diff
@@ -164,10 +164,12 @@ func (pr *Prover) GetLightSignedHeaderAtHeight(height int64) (*tmclient.Header,
 		return nil, err
 	}
 
-	protoVal, err := tmtypes.NewValidatorSet(sh.ValidatorSet.Validators).ToProto()
+	valSet := tmtypes.NewValidatorSet(sh.ValidatorSet.Validators)
+	protoVal, err := valSet.ToProto()
 	if err != nil {
 		return nil, err
 	}
+	protoVal.TotalVotingPower = valSet.TotalVotingPower()
```
Comment (siburu) [inline chains/tendermint/light.go:172]: Does this change fix a bug in the original version?

File: chains/tendermint/prover.go [hunk 2]
```diff
@@ -187,20 +192,19 @@ func (pr *Prover) UpdateLightWithHeader() (header core.HeaderI, provableHeight i
...
 		SignedHeader: sh.SignedHeader.ToProto(),
 		ValidatorSet: protoVal,
 	}
-	queryableHeight = int64(h.GetHeight().GetRevisionHeight())
-	// NOTE: We query connection at height - 1 because of the way tendermint returns
-	// proofs the commit for height n is contained in the header of height n + 1
-	provableHeight = queryableHeight - 1
-	return h, provableHeight, queryableHeight, nil
+	height := int64(h.GetHeight().GetRevisionHeight())
+	return h, height, height, nil
```
Comment (siburu) [inline chains/tendermint/prover.go:207]: Does this modification follow changes of implementation of Tendermint?

File: chains/tendermint/light.go (hunk 1 above)
Comment (bluele) [reply chains/tendermint/light.go:172]: It's required to support tendermint-rs's verifier. The verifier checks if `TotalVotingPower` matches computed one from each power.
https://github.com/informalsystems/tendermint-rs/blob/d102af4430900f27f8113bd8a048bfb22cba3a7d/tendermint/src/validator.rs#L45

File: chains/tendermint/prover.go (hunk 2 above)
Comment (bluele) [reply chains/tendermint/prover.go:207]: yeah, this is due to cosmos-sdk updates: https://github.com/cosmos/cosmos-sdk/pull/9879

**Review by siburu — APPROVED** [review]

**Pull request merged**